File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    while (strs[0] !== "")
    {
        if (strs.filter(x => x.slice(0, strs[0].length).includes(strs[0])).length === strs.length) return strs[0];
        else (strs[0] = strs[0].slice(0, strs[0].length-1));
    }
    return ""
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";

        int charPtr = 0;
        char curChar;
        string result = "";
        bool foundResult = false;

        while (!foundResult) {
            if (charPtr == strs[0].size()) {
                foundResult = true;
                break;
            }
            char curChar = strs[0][charPtr];
            for (const string &s : strs) {
                if (charPtr == s.size() || s[charPtr] != curChar) {
                    foundResult = true;
                    break;
                }
            }
            if (!foundResult) {
                result += curChar;
                ++charPtr;
            } 
        }
        
        return result;
    }
};